modules/seqexec/server/src/main/scala/seqexec/server/gnirs/GnirsEpics.scala

```diff
@@ -126,100 +133,99 @@ class GnirsEpics(epicsService: CaService, tops: Map[String, String]) {
     def setLabel(v: String): SeqAction[Unit] = setParameter(label, v)
   }
 
-  val state: CaStatusAcceptor = epicsService.getStatusAcceptor("nirs::status")
-  val dcState: CaStatusAcceptor = epicsService.getStatusAcceptor("nirs::dcstatus")
-
-  def arrayId: Option[String] = Option(dcState.getStringAttribute("arrayid").value)
+  private val state: CaStatusAcceptor = epicsService.getStatusAcceptor("nirs::status")
+  private val dcState: CaStatusAcceptor = epicsService.getStatusAcceptor("nirs::dcstatus")
 
-  def arrayType: Option[String] = Option(dcState.getStringAttribute("arraytyp").value)
+  def arrayId: F[Option[String]] = safeAttribute(dcState.getStringAttribute("arrayid"))
```

Comment:
We are leaning to produce items with the form `F[A]` rather than `F[Option[A]]` but I want to do than in a later PR